### PR TITLE
fix: remove github token input, refer to secret, add var checks

### PR DESCRIPTION
**Pull request by Glenn-Terjesen**

### CODE REVIEW

File: .github/actions/k8s-auth/action.yml [hunk 1]
```diff
@@ -7,9 +7,6 @@ inputs:
   cloud_provider:
     description: "Which cloud service provider to use - Google Cloud: 'gcp' or Azure: 'az'"
     default: "gcp"
-  github_token:
-    description: "GitHub token"
-    required: true
```
Comment (AlexanderBrevig) [inline .github/actions/k8s-auth/action.yml:12]: This will not work, must be accepted as input. Imagine if a random action could just willy-nilly access all our secrets without being granted explicit access... 🤠

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]

File: .github/actions/k8s-auth/action.yml (hunk 1 above)
Comment (AlexanderBrevig) [reply .github/actions/k8s-auth/action.yml:12]: I should have highlighted https://github.com/entur/gha-meta/pull/34/files#diff-10ca432c426ea26768f0ba50c81305f128d1a267f5a0dfa8339f0d5ff7b678d8R53 `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}` as this is the part that won't work

TLDR: we need this input

Comment (Glenn-Terjesen) [reply .github/actions/k8s-auth/action.yml:12]: This does not compute.. at all :D 
We are basically doing the same as here: https://github.com/weinong/azure-federated-identity-samples/blob/main/.github/workflows/access-aks.yml
We don't need to forward the github_token at all.

Comment (Glenn-Terjesen) [reply .github/actions/k8s-auth/action.yml:12]: https://azure.github.io/kubelogin/concepts/login-modes/workloadidentity.html

Comment (AlexanderBrevig) [reply .github/actions/k8s-auth/action.yml:12]: https://github.com/Azure/use-kubelogin?tab=readme-ov-file#kubelogin-version

We are _not_ doing the same as the workflows you linked, because we go at it via an action, which has no access to the token as opposed to the workflow.

Comment (AlexanderBrevig) [reply .github/actions/k8s-auth/action.yml:12]: We should probably add tests for these actions in `ci.yml` and if we get a ✅ without GITHUB_TOKEN then I'm OK with the `use-kubelogin` documentation being wrong

File: .github/actions/cloud-auth/action.yml [hunk 2]
```diff
@@ -29,6 +29,22 @@ runs:
...
+    - id: verify-vars-is-set-gcp
+      name: Verify GCP environment variables
+      if: inputs.cloud_provider == 'gcp' && (inputs.workload_identity_provider == '' || inputs.service_account == '')
+      shell: bash
+      run: |
+        echo "Missing required environment variables"
+        echo "Verify that 'workload_identity_provider' and 'service_account' are set and you are using the correct environment name"
+        exit 1
+    - id: verify-vars-is-set-az
+      name: Verify GCP environment variables
```
Comment (bvestli) [inline .github/actions/cloud-auth/action.yml:41]:
```suggestion
      name: Verify Azure environment variables
```

**Review by AlexanderBrevig — APPROVED** [review]
Deferred tests https://github.com/entur/gha-meta/issues/35

File: .github/actions/docker-auth/action.yml [hunk 3]
```diff
@@ -36,6 +36,22 @@ runs:
...
+    - id: verify-vars-is-set-gcp
+      name: Verify GCP environment variables
+      if: inputs.cloud_provider == 'gcp' && (inputs.workload_identity_provider == '' || inputs.service_account == '')
+      shell: bash
+      run: |
+        echo "Missing required environment variables"
+        echo "Verify that 'workload_identity_provider' and 'service_account' are set and you are using the correct environment name"
+        exit 1
+    - id: verify-vars-is-set-az
+      name: Verify GCP environment variables
```
Comment (AlexanderBrevig) [inline .github/actions/docker-auth/action.yml:48]:
```suggestion
      name: Verify Azure environment variables
```
